src/backend/expungeservice/expunger/analyze.py

```diff
@@ -118,3 +134,166 @@ def __init__(self, name, dob, cases):
         self.name = name
         self.dob = dob
         self.cases = cases
+
+    def num_charges(self):
+        num = 0
+        for case in self.cases:
+            num += case.num_charges()
+        return num
+
+class ResultCode(enum.Enum):
+    INELIGIBLE = 'Ineligible'
+    ELIGIBLE = 'Eligible'
+    FURTHER_ANALYSIS = 'Further analysis needed'
+    EXAMINE = 'Examine'
+    NO_ACTION = 'Take no action'
+    OPEN_CASE = 'Open Case'
+
+"""
+A class for storing individual analysis steps
+
+Attributes:
+    result: A boolean that's the result of the check
+    check: A string that breifly describes the check
+    check_desc: A string that elaborates the check (optional)
+"""
+Check = collections.namedtuple('Check', 'result check check_desc',
+                               defaults=[None])
+
+class Result(object):
+    """
+    A class for storing analysis results
+
+    Attributes:
+        code: A ResultCode instance.
+        statute: A Statute instance that's associated with arrived result.
+        date: A datetime.date instance that specifies eligibility date.
+        checks: A list of Check instances that describes the analysis done.
+    """
+    def __init__(self, code=None, checks=None,
+                 statute=None, date=None):
+        self.code = code
+        self.checks = checks
+        self.statute = statute
+        self.date = date
```

Comment:
Would this not be the same for all charges except for the most recent?

Reply:
I believe it may be different for different charges. There's a path in type eligibility flow chart that indicates that the charge would be eligible in one year for no-complaint charges while a similar path in time eligibility chart applies only to most recent charge.

Reply:
I agree with you but I think @michaelatcommunitylaw has to clarify this. From UI perspective, it makes sense to put any eligibility that involves time under "time eligibility".

Reply:
Yeah, I'll ask him to clarify tomorrow.